packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -174,6 +189,17 @@ function DashboardLayout(props: DashboardLayoutProps) {
   const [isNavigationFullyExpanded, setIsNavigationFullyExpanded] =
     React.useState(isNavigationExpanded);
 
+  React.useEffect(() => {
+    if (typeof navigationMenuOpen === 'boolean') {
+      setIsNavigationExpanded(navigationMenuOpen);
+      if (navigationMenuOpen) {
```

Comment:
When `navigationMenuOpen` is a boolean, we cannot use the current `isDesktopNavigationExpanded` or `isMobileNavigationExpanded` states the way they're currently being used in this component.

So besides this effect, we need to make it so that all checks that involve `isDesktopNavigationExpanded` or `isMobileNavigationExpanded` check `navigationMenuOpen` in those cases instead.

And every current call to `setIsDesktopNavigationExpanded` or `setIsMobileNavigationExpanded` can be ignored when `navigationMenuOpen` is a boolean, but the appropriate `onNavigationMenuOpen` or `onNavigationMenuClose` should be called instead.

`onNavigationMenuOpen` and `onNavigationMenuClose` should also still be called when `navigationMenuOpen` is `undefined`, however, so that they can be used for other cases.

Your current approach requires an extra render for the effect to run, which should not be necessary, and the `navigationMenuOpen` value can become out of sync with the UI when the internal state is changed.

Hopefully this makes sense and points you in the right direction!